### PR TITLE
Airbrake Plugin API connectivity

### PR DESCRIPTION
**Pull request by moltenice**
## Hey, I just made a Pull Request!

The Airbrake plugin now talks to its API.

Since the Airbrake API handles API Keys via a URL parameter, an `airbrake-backend` plugin will need to be made that the frontend can talk to. This is the next step.

But, in the meantime, except for the code to connect to this Backend, the initial version of this plugin is ready!

You can test this plugin by starting the standalone dev environment and seeing the mock API work or try using a real API (you'll need to disable CORS in your browser first)

Some fun images of the standalone dev app to aid development:
image
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by LightningStairs — APPROVED** [review]

**Review by punitcse — APPROVED** [review]
🍰

File: ADOPTERS.md [hunk 1]
```diff
@@ -1,92 +1,92 @@
...
-| [IKEA IT AB](https://www.ingka.com)                                       | [@bjornramberg](https://github.com/bjornramberg), [@supriyachitale](https://github.com/supriyachitale) | Supporting engineers at scale with self serve access and connecting the dots of our engineering platform and services, enabling product teams to move faster and go further, and unleashing innovation, reuse and co-creation across the organisation. |
+| Organization                                                                | Contact                                                                                                                                                                                                      | Description of Use                                                                                                                                                                                                                                                                                                                                   |
```
Comment (benjdlambert) [inline ADOPTERS.md]: Don't think we need the changes in this file?

File: .changeset/two-elephants-arrive.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-airbrake': minor
+---
+
+API connectivity has added, but currently will only work by running it standalone on a browser with CORS disabled.
```
Comment (benjdlambert) [inline .changeset/two-elephants-arrive.md:5]: Why is this?

File: plugins/airbrake/src/api/index.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export * from './mock';
+export type { AirbrakeApi } from './airbrake-api';
```
Comment (benjdlambert) [inline plugins/airbrake/src/api/index.ts:18]: Prefer camel case for filenames over snake case.

File: plugins/airbrake/src/api/production-api.test.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ProductionAirbrakeApi } from './production-api';
+import nock from 'nock';
```
Comment (benjdlambert) [inline plugins/airbrake/src/api/production-api.test.ts:18]: Prefer using `msw` to mock network requests.

You can read more here: https://backstage.io/docs/architecture-decisions/adrs-adr007

**Review by benjdlambert — COMMENTED** [review]
Nice work! Thanks for this, just a few nits! ⭐

File: .changeset/two-elephants-arrive.md (hunk 2 above)
Comment (moltenice) [reply .changeset/two-elephants-arrive.md:5]: More explanation in the PR description, but essentially this PR was made before the backend was written and until the Backend is merged it won't really work out of the box due to the CORS policies.

File: plugins/airbrake/src/api/index.ts (hunk 3 above)
Comment (moltenice) [reply plugins/airbrake/src/api/index.ts]: Cool, updated

File: plugins/airbrake/src/api/production-api.test.ts (hunk 4 above)
Comment (moltenice) [reply plugins/airbrake/src/api/production-api.test.ts]: Gotcha, using `msw` now :)

**Comment by moltenice** [comment]
@benjdlambert Thanks for reviewing this PR! All your feedback has been addressed hopefully so would appreciate a re-review :)

**Review by Rugvip — APPROVED** [review]
Nice, thanks! 👍 

:shipit:

**Pull request merged**